ecosystem/sep-0045.md

```diff
@@ -0,0 +1,431 @@
+
+Example:
+
+```
+GET https://auth.example.com/?address=CCIBUCGPOHWMMMFPFTDWBSVHQRT4DIBJ7AD6BZJYDITBK2LCVBYW7HUQ&home_domain=example.com
+```
+
+#### Response
+
+##### Success
+
+On success the endpoint must return `200 OK` HTTP status code and a JSON object with the following fields:
+
+- `authorization_entries`: a list of XDR-encoded `SorobanAuthorizationEntry`. It contains an entry for each of the
+  **Client Account**'s signers in addition to a signed entry from the **Home Domain Address** and optionally the
+  **Client Domain Address**. The entries are not guaranteed to be in any specific order.
+
+Each entry's `root_invocation` function is a `contract_fn` with the following with no additional sub invocations:
+
+- `contract_address` is the `WEB_AUTH_CONTRACT_ID` from the **Server**'s `stellar.toml`
+- `function_name` is `web_auth_verify`
+- `args`:
+  - `address` matches the `address` from the request
+  - `home_domain` matches the `home_domain` from the request
+  - `home_domain_address` matches the **Home Domain Address**
+  - `web_auth_domain` matches the **Server**'s domain
+  - `client_domain` matches the **Client**'s domain
+  - `client_domain_address` matches the **Client Domain Address**
+- `network_passphrase`: (optional but recommended) Stellar network passphrase used by the **Server**. This allows a
+  **Client** to verify that it's using the correct passphrase when signing and is useful for identifying when a
+  **Client** or **Server** have been configured incorrectly.
+
+Example:
+
+```json
+{
+  "authorization_entries": [
+    "AAAAAQAAAAHw6CVqzY+dCq3myVJBo1kb3nEGE7oO6obmJeUNvYQ0unUQ/ZB5+1lbAAAAAAAAAAEAAAAAAAAAAahbWC88MmIccL/85Jd6Myxjq5sHF+2dnq1+2H8osKvEAAAAD3dlYl9hdXRoX3ZlcmlmeQAAAAAHAAAAEgAAAAHw6CVqzY+dCq3myVJBo1kb3nEGE7oO6obmJeUNvYQ0ugAAAA4AAAAObG9jYWxob3N0OjgwODAAAAAAABIAAAAAAAAAANKw4wpgrtrVoAuJ7zcXgZAOgF0etbzpRfZJhXKBED5VAAAADgAAAA5sb2NhbGhvc3Q6ODA4MAAAAAAAAQAAAAEAAAAOAAAACTg5MjE5NDQ1MQAAAAAAAAA=",
+    "AAAAAQAAAAAAAAAA0rDjCmCu2tWgC4nvNxeBkA6AXR61vOlF9kmFcoEQPlUEH887Ol7oawAT5vYAAAAQAAAAAQAAAAEAAAARAAAAAQAAAAIAAAAPAAAACnB1YmxpY19rZXkAAAAAAA0AAAAg0rDjCmCu2tWgC4nvNxeBkA6AXR61vOlF9kmFcoEQPlUAAAAPAAAACXNpZ25hdHVyZQAAAAAAAA0AAABA42dUagEASp9894JchR8zl8qIqSVc5KhuWgxOu1YWDH2nLBlwwQXAMX4h5i1zA5Z86LU84aSPoVQtdzCeJZG2BgAAAAAAAAABqFtYLzwyYhxwv/zkl3ozLGOrmwcX7Z2erX7Yfyiwq8QAAAAPd2ViX2F1dGhfdmVyaWZ5AAAAAAcAAAASAAAAAfDoJWrNj50KrebJUkGjWRvecQYTug7qhuYl5Q29hDS6AAAADgAAAA5sb2NhbGhvc3Q6ODA4MAAAAAAAEgAAAAAAAAAA0rDjCmCu2tWgC4nvNxeBkA6AXR61vOlF9kmFcoEQPlUAAAAOAAAADmxvY2FsaG9zdDo4MDgwAAAAAAABAAAAAQAAAA4AAAAJODkyMTk0NDUxAAAAAAAAAA=="
```

Comment:
These auth entries could be wrapped in a variable length array so that they're a single unit to decode. I've noticed in some other APIs where we designed them to be an array of XDR, that it's more annoying to interact with than a single XDR value. This is because you have to iterate over the values and decode each one, rather than decode a single blob.

Reply:
If helpful we can add a new type to the XDR such as:
```
type SorobanAuthorizationEntries SorobanAuthorizationEntry<>;
```

Reply:
I like this idea. Since this will require an XDR change and likely a discussion, what do you think about keeping the auth entries in an array and updating this SEP with the new type later?

Reply:
The XDR change is not protocol related, so it can be made at anytime, and I think is relatively uncontroversial, so I think you could rely on it here. Also, in most SDKs the XDR change is not required, it's just convenient for the CLI that can only decode things with a name. Most SDKs should be able to decode an array of a type as is today.